File: app/src/main/java/com/callmangement/ui/ins_weighing_scale/activity/LocationUtils.java
package com.callmangement.ui.ins_weighing_scale.activity;

import android.content.Context;
import android.location.LocationManager;

public class LocationUtils {
    public static boolean isLocationEnabled(Context context) {
        LocationManager locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
        boolean gpsEnabled = false;
        boolean networkEnabled = false;

        try {
            gpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
        } catch (Exception e) {
            e.printStackTrace();
        }

        try {
            networkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
        } catch (Exception e) {
            e.printStackTrace();
        }

        return gpsEnabled || networkEnabled;
    }
}

File: app/src/main/java/com/callmangement/ui/ins_weighing_scale/activity/LocationUtils.kt
package com.callmangement.ui.ins_weighing_scale.activity

import android.content.Context
import android.location.LocationManager

object LocationUtils {
    fun isLocationEnabled(context: Context): Boolean {
        val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
        var gpsEnabled = false
        var networkEnabled = false

        try {
            gpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
        } catch (e: Exception) {
            e.printStackTrace()
        }

        try {
            networkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return gpsEnabled || networkEnabled
    }
}
